projects/igniteui-angular/src/lib/core/displayDensity.ts

```diff
@@ -77,10 +95,19 @@ export class DisplayDensityBase {
     /**
      *@hidden
      */
-    protected isCompact(): boolean {
+    public isCompact(): boolean {
         return this._displayDensity === DisplayDensity.compact ||
             (!this._displayDensity && this.displayDensityOptions && this.displayDensityOptions.displayDensity === DisplayDensity.compact);
     }
+    constructor(protected displayDensityOptions: IDisplayDensity) {
+        Object.assign(this.oldDisplayDensityOptions, displayDensityOptions);
+    }
 
-    constructor(protected displayDensityOptions: IDisplayDensity) {}
+    public ngDoCheck() {
+        if (this.oldDisplayDensityOptions && this.displayDensityOptions &&
+            this.oldDisplayDensityOptions.displayDensity !== this.displayDensityOptions.displayDensity) {
+            this.oldDisplayDensityOptions = Object.assign(this.oldDisplayDensityOptions, this.displayDensityOptions);
+            this.onDensityChanged.emit();
```

Comment:
@kacheshmarova There are no arguments for this emit. Is this on purpose?

Reply:
It was not on purpose,  I added these arguments.